driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java

```diff
@@ -66,12 +66,13 @@
                         binding.release();
                         return Mono.error(new MongoClientException("Read preference in a transaction must be primary"));
                     } else {
-                        return Mono.<T>create(sink -> operation.executeAsync(binding, sinkToCallback(sink)))
-                                .doOnTerminate(binding::release)
-                                .doOnError((t) -> {
-                                    labelException(session, t);
-                                    unpinServerAddressOnTransientTransactionError(session, t);
-                                });
+                        return Mono.<T>create(sink -> operation.executeAsync(binding, (result, t) -> {
+                            binding.release();
+                            sinkToCallback(sink).onResult(result, t);
```

Comment:
Does anyone else find `syncToCallback` confusing?  It seems more like the reverse, `callbackToSink`, makes more sense.

Reply:
It converts the sink to a SingleResultCallback so makes sense to me.